generated/nidcpower/nidcpower/session.py

```diff
@@ -6271,11 +6275,11 @@ def query_output_state(self, output_state):
             output_state (enums.OutputStates): Specifies the output state of the output channel that is being queried.
                 **Defined Values**:
 
-                +----------------------+-------------------------------------------------------------------+
-                | OutputStates.VOLTAGE | The device maintains a constant voltage by adjusting the current. |
-                +----------------------+-------------------------------------------------------------------+
-                | OutputStates.CURRENT | The device maintains a constant current by adjusting the voltage. |
-                +----------------------+-------------------------------------------------------------------+
+                +-------------------------------+-------------------------------------------------------------------+
+                | OutputStates.CONSTANT_VOLTAGE | The device maintains a constant voltage by adjusting the current. |
+                +-------------------------------+-------------------------------------------------------------------+
+                | OutputStates.CONSTANT_CURRENT | The device maintains a constant current by adjusting the voltage. |
```

Comment:
@marcoskirsch @ni-jfitzger The two enum values here are renamed because new enum values were added to the OutputStates enum. (Side note: I mentioned the renames and additions in the changelog.)

1. This is a breaking change, because the enum values were renamed. Does that need a special call-out?
2. The new enum values are not listed here because they're not listed in the doc table for this particular function parameter (in the metadata). Should they be?

Reply:
>This is a breaking change, because the enum values were renamed. Does that need a special call-out?

Hmm, the new names do seem to match (though not casing) the names used in the LabVIEW API.
https://www.ni.com/docs/en-US/bundle/ni-dcpower-labview-api-ref/page/nidcpowerviref/nidcpower_query_output_state.html

Still, I'm not sure if that meets the bar to justify breaking client code. We have documentation for these enums, if clarification is needed.

I do like the new names better, though. Can we create aliases?

Reply:
Yuck, the codegenerator is "smart" in that it strips the prefix of enum names if they all match. 90% of the time that's a good thing because we "namespace" the C `#define`s by using a common prefix.

In this case, I believe we should **not** have stripped the prefix. But we aren't adding more enum values so for now we should just avoid the source breaker.

Reply:
I think the follow up actions are:

1. change metadata to override this so it remains as it was
2. File an issue tagged with source breaker so we can tackle this when/if we do a source breaking 2.0 release.

Reply:
> But we aren't adding more enum values so for now ...

But we _are_ adding more enum values. We're just not documenting them in this function's parameter docs.

If we intentionally don't want to support those new enum values for this function, then should we introduce a "fancy" function to explicitly reject those new enum values? Should I do something in the hapigen data to erase the added enum values at the python layer?

Alternatively, if we're okay supporting the new enum values, then I should probably just update the function parameter docs to include them.

Reply:
In [!372484](https://ni.visualstudio.com/DevCentral/_git/ni-central/pullrequest/372484), I took the approach of adding aliases for the enum values. Not sure if you'll like it or not.

(And I'm still hoping for an answer on whether or not the other enum values should be permitted -- or perhaps explicitly rejected -- as valid inputs for the `query_output_state` function.)

Reply:
Reverted this change.